### ISSUE TITLE
[FR]: VisionOs

### ISSUE THREAD

**Issue by fishandphil** [firebase/firebase-ios-sdk#12024]
### Description

Good morning

Is it possible to be pointed out to any documentation or thread explaining how to compile a XCFramework for Firebase that is compatible with VisionOS

Thank you

### Firebase Product(s)

Firestore

**Comment by paulb777** [comment]
Thanks for the request. We'll likely include visionOS slices in xcframeworks in 2024. In the meantime, please refer to Apple's documentation.

**Comment by PSchmiedmayer** [comment]
@paulb777 Are there any updates regarding to include visionOS builds in XCFrameworks? With the device launching soon, it would be nice to use Firebase as a dependency on the Apple Vision Pro for native visionOS apps.

CC @Supereg

**Comment by paulb777** [comment]
We're considering the `xcframework` support later in the spring when we drop Xcode 14 support. In the meantime, visionOS is supported via SPM source distributions.

**Comment by PSchmiedmayer** [comment]
Thank you for the answer @paulb777! We will use the source distribution for now but look forward to the binary dependencies in the future once you drop Xcode 14 support 👍

**Comment by rae** [comment]
Ok it's March now and both the device and Xcode 15.3 have been out for a while. Any update?

**Comment by paulb777** [comment]
The App Store continues to support Xcode 14 through April.

**Comment by ygee07** [comment]
Is it possible to build an archive for submission to the App Store using the `open --env ...` and then doing a `Product > Archive`? 

If no, are there any guides we can follow to build an Archive using Firestore as source distribution?

**Comment by paulb777** [comment]
@ygee07 There's nothing special about archiving the source distribution. At the cost of its slow build time, it makes Firestore like every other Firebase product except Analytics.

**Comment by StephenHodgson** [comment]
+1 for tracking

**Comment by paulb777** [comment]
This may be feasible before Firebase 11 since we're planning to start requiring Xcode 15.2 starting with Firebase 10.25.0.

**Comment by arthurschiller** [comment]
+1, would be great to not be forced anymore to always use the --env FIREBASE_SOURCE_FIRESTORE detour.

**Comment by mesqueeb** [comment]
@paulb777 just a quick question, why did you remove it from the Firebase 11 milestone?

**Comment by paulb777** [comment]
The task is more challenging than initially thought - our binary distributions still depend on CocoaPods, which does not yet have good support for VisionOS.  Also, the priority remains lower than many other projects.

**Comment by mesqueeb** [comment]
@paulb777 
My current setup is in `Package.swift`
```swift
dependencies: [
    .package(url: "https://github.com/firebase/firebase-ios-sdk.git", from: "10.28.1"),
]
// ...
dependencies: [.product(name: "FirebaseFirestore", package: "Firebase")]
```
But the downsides are:
1. it takes very long to build even though I only use Firestore in my project.
2. I have to open Xcode with a special terminal command every time.

I imagine 2. will get solved _eventually_, but 1 is giving me the most pain.
**Do you have suggestions how I can speed up Firebase build times?** (especially for cases where we only need to use Firestore)

Thanks in advance.

**Comment by arthurschiller** [comment]
Any news here? The extra work to required to build for visionOS slows down development quite a bit. Not to mention the really long build times.

Thank you! 🙏

**Comment by paulb777** [comment]
Our binary distributions still rely upon CocoaPods, so this feature is dependent upon either CocoaPods support for visionOS or migrating our binary creation process to use SwiftPM.

**Comment by arthurschiller** [comment]
> Our binary distributions still rely upon CocoaPods, so this feature is dependent upon either CocoaPods support for visionOS or migrating our binary creation process to use SwiftPM.

Got it! How likely is one over the other though? And how much effort would require option 2 require?

**Comment by DatBlaueHus** [comment]
I'm trying to build from source, but get a build failure:
'absl/strings/escaping.h' file not found

Any idea how to address that?

**Comment by arthurschiller** [comment]
Are there any new developments in this area?

**Comment by arthurschiller** [comment]
Bumping this up yet again.

**Comment by ncooke3** [comment]
Hi @arthurschiller, no developments to share. Building our binaries with SPM is complex and requires more investigation.

**Comment by arthurschiller** [comment]
Bump… 🥹😮‍💨 I don't wanna rant but it's been almost 2 years since visionOS was released. In Beta since June 2023.
If visionOS is not a priority for Google – fine, but then please just communicate it clearly so we know what to expect. I am wasting a lot of time when switching back and forth between visionOS and iOS/macOS builds.

I think for a huge company like Google it should be possible to either provide visionOS XCFramework binaries or to provide a proper workaround that works in automated build environments.

**Comment by arthurschiller** [comment]
For anyone blocked on this — I've published a working overlay package:
                                                                                                           
https://github.com/arthurschiller/firebase-firestore-xcframeworks
                                                                                                           
It ships pre-built XCFrameworks for the 6 Firestore-related binaries (FirebaseFirestoreInternal, grpc,     
grpcpp, openssl_grpc, absl, leveldb) with visionOS slices added alongside Google's untouched               
iOS/macOS/Catalyst/tvOS slices. Consumers add it as a single SPM dependency and import                     
FirebaseFirestorePrebuilt in place of FirebaseFirestore — no source compile, no FIREBASE_SOURCE_FIRESTORE
env var, links cleanly on Xcode Cloud including visionOS.

Currently tracks Firebase 11.15.x. New Firebase versions can be re-spun via a one-command GitHub Actions   
workflow.
                                                                                                           
Not a replacement for official support — still very much hoping Google ships first-party visionOS binaries 
— but might unblock people in the meantime.

**Comment by StephenHodgson** [comment]
pretty sure it ain't coming 😏 But thank you so much @arthurschiller!

**Comment by arthurschiller** [comment]
> pretty sure it ain't coming 😏 But thank you so much [@arthurschiller](https://github.com/arthurschiller)!

Felt the same, so kinda lost patience haha. It's a huge QoL win to have a single Xcode Claude workflow for all platforms.

**Comment by ncooke3** [comment]
Hi everyone, apologies for the delay. We have plans to modernize our XCFramework build pipeline. Pre-built visionOS framework slices would be in scope for support. Our current XCFramework tooling has dependencies on the CocoaPods build system, and with CocoaPods going away, I'll be working modernizing our tooling.

In the meantime, thank you for you patience and feedback!

**Comment by arthurschiller** [comment]
> Hi everyone, apologies for the delay. We have plans to modernize our XCFramework build pipeline. Pre-built visionOS framework slices would be in scope for support. Our current XCFramework tooling has dependencies on the CocoaPods build system, and with CocoaPods going away, I'll be working modernizing our tooling.
> 
> In the meantime, thank you for you patience and feedback!

That's great news! Appreciate the update @ncooke3!
